tests/Mono.Android-Tests/jni/reuse-threads.c

```diff
@@ -200,3 +200,46 @@ rt_invoke_callback_on_new_thread (CB cb)
 	return 0;
 }
 
+static int
+_register_type_from_new_thread (const char *java_type_name, jobject class_loader)
+{
+	JNIEnv *env              = _get_env ("_register_type_from_new_thread");
+	jclass ClassLoader_class = (*env)->FindClass (env, "java/lang/ClassLoader");
+	jmethodID loadClass      = (*env)->GetMethodID (env, ClassLoader_class, "loadClass", "(Ljava/lang/String;)Ljava/lang/Class;");
+	jobject loaded_class     = (*env)->CallObjectMethod (env, class_loader, loadClass, java_type_name);
+
+	if ((*env)->ExceptionOccurred (env) != NULL) {
+		__android_log_print (ANDROID_LOG_INFO, "XA/RuntimeTest", "FAILURE: class '%s' cannot be loaded, Java exception thrown!", java_type_name);
+		return -1;
```

Comment:
We should call `(*env)->ExceptionClear(env)` before returning, so that there isn't a pending exception within the unit tests.

Reply:
You may want to use different negative return values, so that it's more obvious in unit test output which part failed.